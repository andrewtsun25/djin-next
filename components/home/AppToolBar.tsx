import React from "react";
import { AppBar, Box, IconButton, Toolbar, Typography } from "@mui/material";
import MenuIcon from "@mui/icons-material/Menu";
import Link from "next/link";
import logo from "../../public/logos/logo.png";
import Image from "next/image";

interface AppToolBarProps {
  isAppDrawerOpen: boolean;
  setIsAppDrawerOpen(newIsDrawerOpen: boolean): void;
}

export const MENU_BUTTON_ROLE = "switch";

const AppToolBar: React.FC<AppToolBarProps> = ({
  isAppDrawerOpen,
  setIsAppDrawerOpen,
}: AppToolBarProps) => {
  return (
    <AppBar position="sticky">
      <Toolbar>
        <IconButton
          color="inherit"
          aria-label="open drawer"
          role={MENU_BUTTON_ROLE}
          onClick={() => setIsAppDrawerOpen(true)}
          edge="start"
          sx={{ mr: 2, ...(isAppDrawerOpen && { display: "none" }) }}
        >
          <MenuIcon />
        </IconButton>
        <Link href="/">
          <Box height="100%" display="flex" alignItems="center">
            <Image src={logo} alt="d.jin website logo" width={50} height={50} />
            <Typography variant="h6" ml={1}>
              d.jin
            </Typography>
          </Box>
        </Link>
      </Toolbar>
    </AppBar>
  );
};

export default AppToolBar;
