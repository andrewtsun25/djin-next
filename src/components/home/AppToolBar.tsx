import MenuIcon from "@mui/icons-material/Menu";
import {
  AppBar,
  IconButton,
  Toolbar,
  useMediaQuery,
  useTheme,
} from "@mui/material";
import Link from "next/link";
import { useRouter } from "next/router";
import React, { useMemo } from "react";

import logo from "../../../public/logos/logo.png";
import {
  AppToolBarBrandContainer,
  AppToolBarLogo,
  HomeLinkText,
} from "./styled";

interface AppToolBarProps {
  isAppDrawerOpen: boolean;
  setIsAppDrawerOpen(newIsDrawerOpen: boolean): void;
}

export const MENU_BUTTON_ROLE = "switch";

const pathsWithFixedAppToolbar = ["/"];

const AppToolBar: React.FC<AppToolBarProps> = ({
  isAppDrawerOpen,
  setIsAppDrawerOpen,
}: AppToolBarProps) => {
  const router = useRouter();
  const theme = useTheme();
  const isDesktop = useMediaQuery(theme.breakpoints.up("md"));
  const isAppToolbarPositionFixed = useMemo(
    () => pathsWithFixedAppToolbar.some((path) => router.asPath === path),
    [router],
  );
  return (
    <AppBar position={isAppToolbarPositionFixed ? "fixed" : "sticky"}>
      <Toolbar>
        {isDesktop && (
          <IconButton
            color="inherit"
            aria-label="open drawer"
            role={MENU_BUTTON_ROLE}
            onClick={() => setIsAppDrawerOpen(true)}
            edge="start"
            sx={{ mr: 2, ...(isAppDrawerOpen && { display: "none" }) }}
          >
            <MenuIcon />
          </IconButton>
        )}
        <Link href="/" style={{ textDecoration: "none" }}>
          <AppToolBarBrandContainer>
            <AppToolBarLogo src={logo} alt="d.jin website logo" />
            <HomeLinkText variant="h5">d.jin</HomeLinkText>
          </AppToolBarBrandContainer>
        </Link>
      </Toolbar>
    </AppBar>
  );
};

export default AppToolBar;
