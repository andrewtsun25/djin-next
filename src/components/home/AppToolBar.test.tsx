import { fireEvent, render } from "@testing-library/react";
import AppToolBar, { MENU_BUTTON_ROLE } from "./AppToolBar";

describe("AppToolBar", () => {
  let setIsAppDrawerOpen: (b: boolean) => void;

  beforeEach(() => {
    setIsAppDrawerOpen = jest.fn();
  });

  it("renders", () => {
    const isAppDrawerOpen = true;
    const appToolBar = render(
      <AppToolBar
        isAppDrawerOpen={isAppDrawerOpen}
        setIsAppDrawerOpen={setIsAppDrawerOpen}
      />
    );
    expect(appToolBar).toMatchSnapshot();
  });

  it("opens the drawer if the drawer isn't opened", () => {
    const isAppDrawerOpen = false;
    const { getByRole } = render(
      <AppToolBar
        isAppDrawerOpen={isAppDrawerOpen}
        setIsAppDrawerOpen={setIsAppDrawerOpen}
      />
    );
    const menuButton = getByRole(MENU_BUTTON_ROLE);
    fireEvent.click(menuButton);
    expect(setIsAppDrawerOpen).toHaveBeenCalledWith(true);
  });

  it("doesn't render the menu button if the drawer is open", () => {
    const isAppDrawerOpen = true;
    const { queryByRole } = render(
      <AppToolBar
        isAppDrawerOpen={isAppDrawerOpen}
        setIsAppDrawerOpen={setIsAppDrawerOpen}
      />
    );
    const menuButton = queryByRole(MENU_BUTTON_ROLE);
    expect(menuButton).toBeNull();
  });
});
