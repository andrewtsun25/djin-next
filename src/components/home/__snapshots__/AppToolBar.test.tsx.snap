// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppToolBar renders 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <header
        class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionFixed mui-fixed css-11b3ww9-MuiPaper-root-MuiAppBar-root"
      >
        <div
          class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-hyum1k-MuiToolbar-root"
        >
          <button
            aria-label="open drawer"
            class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-edgeStart MuiIconButton-sizeMedium css-13k3028-MuiButtonBase-root-MuiIconButton-root"
            role="switch"
            tabindex="0"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="MenuIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
              />
            </svg>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
          <a
            href="/"
          >
            <div
              class="MuiBox-root css-15c13yl"
            >
              <img
                alt="d.jin website logo"
                data-nimg="1"
                decoding="async"
                height="50"
                loading="lazy"
                src="/_next/image?url=%2Fimg.jpg&w=128&q=75"
                srcset="/_next/image?url=%2Fimg.jpg&w=64&q=75 1x, /_next/image?url=%2Fimg.jpg&w=128&q=75 2x"
                style="color: transparent;"
                width="50"
              />
              <h5
                class="MuiTypography-root MuiTypography-h5 css-109s1dp-MuiTypography-root"
              >
                d.jin
              </h5>
            </div>
          </a>
        </div>
      </header>
    </div>
  </body>,
  "container": <div>
    <header
      class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionFixed mui-fixed css-11b3ww9-MuiPaper-root-MuiAppBar-root"
    >
      <div
        class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-hyum1k-MuiToolbar-root"
      >
        <button
          aria-label="open drawer"
          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-edgeStart MuiIconButton-sizeMedium css-13k3028-MuiButtonBase-root-MuiIconButton-root"
          role="switch"
          tabindex="0"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="MenuIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
            />
          </svg>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
        <a
          href="/"
        >
          <div
            class="MuiBox-root css-15c13yl"
          >
            <img
              alt="d.jin website logo"
              data-nimg="1"
              decoding="async"
              height="50"
              loading="lazy"
              src="/_next/image?url=%2Fimg.jpg&w=128&q=75"
              srcset="/_next/image?url=%2Fimg.jpg&w=64&q=75 1x, /_next/image?url=%2Fimg.jpg&w=128&q=75 2x"
              style="color: transparent;"
              width="50"
            />
            <h5
              class="MuiTypography-root MuiTypography-h5 css-109s1dp-MuiTypography-root"
            >
              d.jin
            </h5>
          </div>
        </a>
      </div>
    </header>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
